### PR TITLE
[HACK Week] Order Navigation

### PR DESCRIPTION
**Pull request by toupper**
## Description

With this PR we implement order navigation within the order details. With this, users can navigate to previous or next orders without the need to go back to the order list and tap on an order again. This was requested a couple of times through AppStore/Google Play reviews.
If the Order Detail screen is presented through a notification or shortcut these buttons are hidden.

### Known caveats
With this implementation, the user can only visit those orders that were retrieved when loading the order list. That means that if there's only one page loaded the user will be bound to those orders. Even with that and given the size of our pages (25) the users can still benefit from it.

## Testing instructions

1. Go to orders
2. Tap on an order
3. See that the open order is the one that was selected previously
4. See that you can navigate to previous and next orders, but when you reach a bound -upper or lower- the matching button is disabled.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/1864060/652950a4-a906-4a1b-9c8e-39de934a66c1

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift [hunk 1]
```diff
@@ -50,16 +50,29 @@ final class OrderDetailsViewController: UIViewController {
...
+
+    private var viewModel: OrderDetailsViewModel {
+        viewModels[currentIndex]
+    }
 
     private let notices = OrderDetailsNotices()
 
+    private let currentIndex: Int
+
+    private lazy var isSplitViewInOrdersTabEnabled: Bool = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.splitViewInOrdersTab)
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift]: Does need to be `lazy var`?
```suggestion
    private let isSplitViewInOrdersTabEnabled: Bool = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.splitViewInOrdersTab)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift [hunk 2]
```diff
@@ -140,7 +150,51 @@ private extension OrderDetailsViewController {
...
+
+    @objc func loadNextOrder() {
+        loadOrder(with: currentIndex + 1)
+    }
+
+    func loadOrder(with index: Int) {
+        guard let navigationController = isSplitViewInOrdersTabEnabled ?
+                splitViewController?.viewControllers.first as? UINavigationController : navigationController else {
+            return
+        }
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift]: I find this bit hard to follow, could be more readable with some sort switch statement, or declaring/assigning the variables locally?

File: WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift [hunk 3]
```diff
@@ -101,7 +101,7 @@ final class OrderListViewController: UIViewController, GhostableViewController {
 
     /// Callback closure when an order is selected
     ///
-    private var switchDetailsHandler: (OrderDetailsViewModel?) -> Void
+    private var switchDetailsHandler: ([OrderDetailsViewModel], Int) -> Void
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift]: What if we add some names to the tuple parameters? Could make it easier to read and use later on, something like:

```suggestion
    private var switchDetailsHandler: (_ allViewModels: [OrderDetailsViewModel], _ index: Int) -> Void
```

File: WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift [hunk 4]
```diff
@@ -692,11 +692,16 @@ extension OrderListViewController: UITableViewDelegate {
...
         selectedOrderID = order.orderID
+        let allViewModels = allViewModels()
+        let currentIndex = allViewModels.firstIndex(where: { $0.order.orderID == order.orderID })
+
+        guard let currentIndex = currentIndex else { return }
 
         if isSplitViewInOrdersTabEnabled {
-            switchDetailsHandler(orderDetailsViewModel)
+            splitViewController?.isCollapsed ?? true ? switchDetailsHandler(allViewModels, currentIndex) :
+            switchDetailsHandler([orderDetailsViewModel], 0)
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift]: Same as above, the logic here is hard to follow 😅 , could we make it more readable with some switch or re-declaring these locally?

**Review by iamgabrielma — APPROVED** [review]
Nice improvement! LGTM :shipit:

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift (hunk 2 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift]: Sure, I tried to make it more readable in f257f1612901127ec291c105b4086210a3e216ab

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift (hunk 1 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift]: Done in f3d0e15eaf6f0b01e36d53cdcc395321d153d896

File: WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift (hunk 3 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift]: Good idea! Done in 40c1dfc8bdc3297b61b8ffbefbb6e3eea8317e94

File: WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift (hunk 4 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift]: Yeah, keeping both the split view logic and single view is tricky, I made it more readable with comments in 8af2c0033d2441508d7c90da9b082a2508f0f84d
